torch/csrc/Size.cpp

```diff
@@ -148,8 +148,33 @@ static PyObject *THPSize_numel(THPSize *self)
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject *THPSize_reduce(THPSize *self)
+{
+  HANDLE_TH_ERRORS
+  auto ret = THPObjectPtr{PyTuple_New(2)};
+  if (!ret) throw python_error();
+
+  auto module = THPObjectPtr(PyImport_ImportModule("torch"));
+  if (!module) throw python_error();
+
+  THPObjectPtr obj(PyObject_GetAttrString(module, "Size"));
+  if (!obj) throw python_error();
+  PyTuple_SET_ITEM(ret.get(), 0, obj.release());
+
+  THPObjectPtr t(PyTuple_New(PyTuple_Size((PyObject*)self)));
+  if (!t) throw python_error();
+  for (Py_ssize_t i = 0; i < PyTuple_Size((PyObject*)self); ++i) {
+    PyTuple_SET_ITEM(t.get(), i, PyTuple_GET_ITEM(self, i));
+  }
+  PyTuple_SET_ITEM(ret.get(), 1, Py_BuildValue("(O)", t.release()));
```

Comment:
I think this should be `t.get()` instead of `t.release()`, otherwise it leaks `t`. (Py_BuildValue has normal reference counting semantics -- it increments the reference count of the value).

I think you should also check the return value of Py_BuildValue here before calling PyTuple_SET_ITEM.